### PR TITLE
fix kubectl set subject --all option invalid bug

### LINKED ISSUE #64182: kubectl set subject --all bug
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
 /kind bug
> /kind feature

**What happened**:
kubectl set subject is designed to set the user/serviceaccount in rolebinding/clusterrolebinding object.
However, the --all option is invalid. For example:
`kubectl set subject rolebinding --user=foo --namespace=foo-ns  --all`
Command above should update every rolebinding object in namespace foo-ns, but it would update just only one.

### PR DESCRIPTION
**Pull request by WanLinghao**
**What this PR does / why we need it**:
Fix the kubectl set subject --all option invalid bug and add test cases for it
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #64182

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/set/set_subject.go [hunk 1]
```diff
@@ -262,7 +262,9 @@ func (o *SubjectOptions) Run(fn updateSubjects) error {
 			continue
 		}
 
-		return o.PrintObj(actual, o.Out)
+		if err := o.PrintObj(actual, o.Out); err != nil {
+			return err
```
Comment (CaoShuFeng) [inline pkg/kubectl/cmd/set/set_subject.go]: Why not:
```
allErrs = append(allErrs, err)
```

After this change `kubectl set` will have the same behavior with kubectl 1.10, right?

Comment (WanLinghao) [reply pkg/kubectl/cmd/set/set_subject.go]: @CaoShuFeng We got couple of similar process in other files. I think we'd better keep same pace  with them.For example:
https://github.com/kubernetes/kubernetes/blob/23d9a48e6a59bc3e4faa84a7cf1416853521e60c/pkg/kubectl/cmd/set/set_image.go#L291

Comment (liggitt) [reply pkg/kubectl/cmd/set/set_subject.go]: agree on aggregating errors rather than short-circuiting

**Comment by WanLinghao** [comment]
@liggitt  Done PTAL

File: pkg/kubectl/cmd/set/set_subject.go [hunk 2]
```diff
@@ -262,7 +262,9 @@ func (o *SubjectOptions) Run(fn updateSubjects) error {
 			continue
 		}
 
-		return o.PrintObj(actual, o.Out)
+		if err := o.PrintObj(actual, o.Out); err != nil {
```
Comment (liggitt) [inline pkg/kubectl/cmd/set/set_subject.go]: same return error exists above in the DryRun block

Comment (WanLinghao) [reply pkg/kubectl/cmd/set/set_subject.go]: @liggitt I fix this in #64242

**Comment by WanLinghao** [comment]
@liggitt  PTAL

**Pull request merged**